Mathlib/NumberTheory/NumberField/CanonicalEmbedding/Basic.lean

```diff
@@ -1157,6 +1208,27 @@ theorem normAtAllPlaces_normAtAllPlaces (x : mixedSpace K) :
     normAtAllPlaces (mixedSpaceOfRealSpace (normAtAllPlaces x)) = normAtAllPlaces x :=
   normAtAllPlaces_mixedSpaceOfRealSpace fun _ ↦ (normAtAllPlaces_nonneg _ _)
 
+theorem normAtAllPlaces_norm_at_real_places (x : mixedSpace K) :
+    normAtAllPlaces (fun w ↦ ‖x.1 w‖, x.2) = normAtAllPlaces x := by
+  ext w
+  obtain hw | hw := isReal_or_isComplex w
+  · simp_rw [normAtAllPlaces, normAtPlace_apply_of_isReal hw, norm_norm]
+  · simp_rw [normAtAllPlaces, normAtPlace_apply_of_isComplex hw]
+
+theorem normAtComplexPlaces_normAtAllPlaces (x : mixedSpace K) :
+    normAtComplexPlaces (mixedSpaceOfRealSpace (normAtAllPlaces x)) = normAtAllPlaces x :=
+   normAtComplexPlaces_mixedSpaceOfRealSpace fun _ _ ↦ (normAtAllPlaces_nonneg _ _)
+
+theorem normAtAllPlaces_eq_of_normAtComplexPlaces_eq {x y : mixedSpace K}
```

Comment:
This file is getting huge... do you think it is possible to split it (in another PR)?

Reply:
That's true. I'll put that on my TODO list. Thanks!